server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -258,7 +258,7 @@ private void sendClusterStateToNode(ClusterState clusterState, BytesReference by
             //  and not log an error if it arrives after the timeout
             // -> no need to compress, we already compressed the bytes
             final TransportRequestOptions options = TransportRequestOptions.builder()
-                .withType(TransportRequestOptions.Type.STATE).withCompress(false).build();
+                .withType(TransportRequestOptions.Type.STATE).build();
```

Comment:
I know this is existing, but I think we can lift this up to a singleton so that we do not create a new instance on every publish to every node.